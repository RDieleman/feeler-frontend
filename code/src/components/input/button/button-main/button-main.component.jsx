import React from "react";
import "./button-main.styles.css";
import "../button.styles.css";

export const ButtonMainComponent = ({content, iconLocation, handleOnClick}) =>{
    // Main button
    return(
        <button className="text-body-sec button-main " onClick={() => handleOnClick()}>
            {content}
            {/*Add icon if location is included*/}
            {(iconLocation === undefined)? "" :
                // todo: Replace with svg and set color in css file
                <img src={iconLocation} alt="btn-icon"/>}
        </button>
    )
}