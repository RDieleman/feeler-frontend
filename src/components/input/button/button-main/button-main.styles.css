.button-main{
    background-color: var(--color-main-medium);
    border: var(--border-width) solid var(--color-main-medium);
    border-radius: var(--border-radius);

    text-transform: none;

    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;

    color: var(--color-gen-light);
}

.button-main > div:first-child{
    flex: 1;
}

.button-main > img{
    height: 100%;
    width: auto;;
}