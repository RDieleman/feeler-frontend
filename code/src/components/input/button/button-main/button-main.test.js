import React from "react";
import {shallow} from "enzyme";
import {ButtonMainComponent} from "./button-main.component";

it("Expect to render ButtonMain component", () => {
    expect(shallow(
        <ButtonMainComponent
            handleOnClick={() => "ButtonMain clicked"}
            content="ButtonMain"
            iconLocation="test"
        />)).toMatchSnapshot();
});