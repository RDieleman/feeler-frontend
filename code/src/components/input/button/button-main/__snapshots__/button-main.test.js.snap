// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render ButtonMain component 1`] = `ShallowWrapper {}`;
